_posts/2024-11-19-release-notes-3.6.0.md

```diff
@@ -0,0 +1,207 @@
+---
+category: announcement
+permalink: /news/3.6.0
+title: "Scala 3.6.0 is now available!"
+---
+
+<!-- TODO: Headline image -->
+
+We're happy to announce the next minor release of Scala - 3.6.0 is finally out!
+
+# What’s new in 3.6.0?
+
+Besides multiple bugfixes, this release stabilises multiple experimental features introduced to Scala language after careful review and acceptance by the [Scala Improvement Proposal's Commitee](https://docs.scala-lang.org/sips/). Many of these changes can have a significant impact on the Scala syntax and are introducing new amazing possibilities in writing concise, typesafe as well as easier, and easier to maintain code.
+
+## SIP-47 - Clause Interleaving
+
+The first major feature we're going to cover is the [clause interleaving](https://docs.scala-lang.org/sips/clause-interleaving.html).
+With this change to the language, you're able to define multiple type parameter lists or place them after the first arguments list. Clause interleaving would benefit the path-dependent API creators.
+It would eliminate the need for intermediate representations introducing runtime overhead or usage of complicated polymorphic functions.
+
+```scala
+trait Key { type Value }
+trait DB {
+  def getOrElse(k: Key)[V >: k.Value](default: V): V // dependent type parameter
+}
+```
+
+## SIP-58 - Named Tuples
+
+Another stabilized feature in this release are the Named Tuples. These have been introduced as experimental in Scala 3.5.0 and allowed you to give meaningful names to tuple elements and use those names during constructing, destructuring, and pattern matching.
+
+```scala
+  type Point = (x: Int, y: Int)
+  val point: Point = (x = 1, y = 2)
+  point match
+    case (x = real, y = 0) => println(s"Real number: $real")
+    case _ => println("Point doesn't represent a real number")
+```
+
+This change also introduces improvements to extractors of case classes. You can now define named extractors for a selection of fields, allowing you to unclutter your code from unused variables.
+
+```scala
+case class User(id: Int, name: String, surname: String)
+
+extension (values: Seq[User])
+  //  Collect user IDs of every entry that has the name matching argument
+  def idsWithName(name: String) = values.collect:
+    case User(name = `name`, id = userId) => userId
+```
```

Comment:
I think it would be pretty critical to point out that named tuples now let you compute structural types without needing macros:

https://dotty.epfl.ch/docs/reference/other-new-features/named-tuples.html#computed-field-names-1

which is pretty cool for a lot of usecases - such as query DSL or typed data frames

Reply:
it opens up a new paradigm for metaprogramming

Reply:
I've added a section about this part